node/malus/src/variants/common.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implements common code for nemesis. Currently, only `FakeValidationResult`
+//! interceptor is implemented.
+
+#![allow(missing_docs)]
+
+// Filter wrapping related types.
+use crate::{interceptor::*, shared::MALUS, FakeCandidateValidation};
+use polkadot_node_primitives::{InvalidCandidate, ValidationResult};
+use polkadot_node_subsystem::messages::CandidateValidationMessage;
+
+#[derive(Clone, Debug)]
+/// An interceptor which fakes validation result with a preconfigured result.
+/// Replaces `CandidateValidationSubsystem`.
+pub struct ReplaceValidationResult {
+	fake_backing_validation: Option<FakeCandidateValidation>,
+	fake_approval_validation: Option<FakeCandidateValidation>,
```

Comment:
Default `false` should be sufficient?

Reply:
Or defining a tristate enum

Reply:
There is also the option of combining these two into a single enum:
- Disabled
- BackingInvalid
- BakingAndApprovalInvalid
- ApprovalInvalid

Naming scheme is not the best but IMO it is not ambiguous at least.